misk-hibernate/src/main/kotlin/misk/hibernate/BulkShardMigrator.kt

```diff
@@ -0,0 +1,435 @@
+package misk.hibernate
+
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableSet
+import misk.logging.getLogger
+import org.hibernate.SessionFactory
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.SQLException
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.stream.Collectors
+import java.util.stream.Collectors.joining
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.isSubclassOf
+
+class BulkShardMigrator<R : DbRoot<R>, C : DbChild<R, C>> private constructor(
```

Comment:
docs please! What's this do? How does it relate to sql migrations? What is the API for this class?

Reply:
sure, added some docs